### PR TITLE
Move `sunpy.physics.solar_rotation` to `sunkit-image.coalignment`

### PR DESCRIPTION
**Pull request by wtbarnes**
## PR Description

See discussion in sunpy/sunpy#5965.

## PR Checklist

- [ ] I have followed the guidelines in the [Contributing document](https://docs.sunpy.org/en/latest/dev_guide/contents/newcomers.html)
- [ ] Changes follow the coding style of this project
- [ ] Changes have been formatted and linted
- [ ] Changes pass pytest style unit tests (and `pytest` passes).
- [ ] Changes include any required corresponding changes to the documentation
- [ ] Changes have been rebased and squashed to the minimal number (typically 1) of relevant commits
- [ ] Changes have a descriptive commit message with a short title
- [ ] I have added a `Fixes #XXXX -` or `Closes #XXXX -` comment to auto-close the issue that your PR addresses

### CODE REVIEW

File: sunkit_image/coalignment.py [hunk 1]
```diff
@@ -25,8 +25,11 @@
 import astropy.units as u
 import sunpy.map
 from sunpy.map.mapbase import GenericMap
+from sunpy.physics.differential_rotation import solar_rotate_coordinate
 from sunpy.util.exceptions import warn_user
 
+__author__ = "J. Ireland"
```
Comment (nabobalis) [inline sunkit_image/coalignment.py:31]:
```suggestion
```

File: sunkit_image/coalignment.py [hunk 2]
```diff
@@ -567,3 +572,136 @@ def mapsequence_coalign_by_match_template(
...
+        The input mapsequence.
+    layer_index : int
+        The index layer.  Shifts are calculated relative to the time of
+        this layer.
+    ``**kwargs``
+        These keywords are passed to the function
+        `sunpy.physics.differential_rotation.solar_rotate_coordinate`.
+    Returns
+    -------
+    x, y : `~astropy.units.Quantity`, ~astropy.units.Quantity`
```
Comment (nabobalis) [inline sunkit_image/coalignment.py:608]:
```suggestion
    `~astropy.units.Quantity`, ~astropy.units.Quantity`
```

File: sunkit_image/coalignment.py [hunk 3]
```diff
@@ -567,3 +572,136 @@ def mapsequence_coalign_by_match_template(
...
+        The shifts relative to the index layer that can be applied
+        to the input mapsequence in order to compensate for solar rotation.
+        The shifts are given in arcseconds as understood in helioprojective
+        coordinates systems.
+    """
+    # Size of the data
+    nt = len(mc.maps)
+
+    # Storage for the shifts in arcseconds
```
Comment (nabobalis) [inline sunkit_image/coalignment.py:617]:
```suggestion
```

File: sunkit_image/coalignment.py [hunk 4]
```diff
@@ -567,3 +572,136 @@ def mapsequence_coalign_by_match_template(
...
+    clip : bool
+        If True, then clip off x, y edges in the datasequence that are potentially
+        affected by edges effects.
+    ``**kwargs``
+        These keywords are passed to the function
+        `sunpy.physics.solar_rotation.calculate_solar_rotate_shift`.
+
+    Returns
+    -------
+    output : `sunpy.map.MapSequence`
```
Comment (nabobalis) [inline sunkit_image/coalignment.py:672]:
```suggestion
    `sunpy.map.MapSequence`
```

File: sunkit_image/coalignment.py [hunk 5]
```diff
@@ -567,3 +572,136 @@ def mapsequence_coalign_by_match_template(
...
+    >>> from sunpy.physics.solar_rotation import mapsequence_solar_derotate
+    >>> map1 = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)  # doctest: +REMOTE_DATA
+    >>> map2 = sunpy.map.Map(sunpy.data.sample.EIT_195_IMAGE)  # doctest: +REMOTE_DATA
+    >>> mc = sunpy.map.Map([map1, map2], sequence=True)  # doctest: +REMOTE_DATA
+    >>> derotated_mc = mapsequence_solar_derotate(mc)  # doctest: +SKIP
+    >>> derotated_mc = mapsequence_solar_derotate(mc, layer_index=-1)  # doctest: +SKIP
+    >>> derotated_mc = mapsequence_solar_derotate(mc, clip=False)  # doctest: +SKIP
+    """
+
+    # Size of the data
```
Comment (nabobalis) [inline sunkit_image/coalignment.py:687]:
```suggestion
```

File: sunkit_image/coalignment.py [hunk 6]
```diff
@@ -567,3 +572,136 @@ def mapsequence_coalign_by_match_template(
...
+    nt = len(mc.maps)
+
+    # Storage for the pixel shifts and the shifts in arcseconds
```
Comment (nabobalis) [inline sunkit_image/coalignment.py:690]:
```suggestion
```

File: sunkit_image/coalignment.py [hunk 7]
```diff
@@ -567,3 +572,136 @@ def mapsequence_coalign_by_match_template(
...
+    xshift_keep = np.zeros(nt) * u.pix
+    yshift_keep = np.zeros_like(xshift_keep)
+
+    # If no shifts are passed in, calculate them.  Otherwise,
+    # use the shifts passed in.
```
Comment (nabobalis) [inline sunkit_image/coalignment.py:695]:
```suggestion
```

File: sunkit_image/coalignment.py [hunk 8]
```diff
@@ -567,3 +572,136 @@ def mapsequence_coalign_by_match_template(
...
+    if shift is None:
+        shift = calculate_solar_rotate_shift(mc, layer_index=layer_index, **kwargs)
+    xshift_arcseconds = shift["x"]
+    yshift_arcseconds = shift["y"]
+
+    # Calculate the pixel shifts
```
Comment (nabobalis) [inline sunkit_image/coalignment.py:701]:
```suggestion
```

File: sunkit_image/coalignment.py [hunk 9]
```diff
@@ -567,3 +572,136 @@ def mapsequence_coalign_by_match_template(
...
+    for i, m in enumerate(mc):
+        xshift_keep[i] = xshift_arcseconds[i] / m.scale[0]
+        yshift_keep[i] = yshift_arcseconds[i] / m.scale[1]
+
+    # Apply the pixel shifts and return the mapsequence
```
Comment (nabobalis) [inline sunkit_image/coalignment.py:706]:
```suggestion
```

**Comment by wtbarnes** [comment]
Huh. The CI is not running...

File: sunkit_image/coalignment.py [hunk 10]
```diff
@@ -567,3 +572,136 @@ def mapsequence_coalign_by_match_template(
...
+            continue
+
+        # Calculate the rotation of the center of the map 'm' at its
+        # observation time to the observation time of the reference layer
+        # indicated by "layer_index".
+        new_coordinate = solar_rotate_coordinate(
+            m.center, observer=rotate_to_this_layer.observer_coordinate, **kwargs
+        )
+
+        # Calculate the shift in arcseconds
```
Comment (nabobalis) [inline sunkit_image/coalignment.py:637]:
```suggestion
```

File: sunkit_image/tests/test_coalignment.py [hunk 11]
```diff
@@ -387,3 +392,90 @@ def test_apply_shifts(aia171_test_map):
...
+    m2header = deepcopy(aia171_test_submap.meta)
+    m2header["date-obs"] = "2011-02-15T01:00:00.34"
+    m2 = sunpy.map.Map((aia171_test_submap.data, m2header))
+    m3header = deepcopy(aia171_test_submap.meta)
+    m3header["date-obs"] = "2011-02-15T02:00:00.34"
+    m3 = sunpy.map.Map((aia171_test_submap.data, m3header))
+    return sunpy.map.Map([aia171_test_submap, m2, m3], sequence=True)
+
+
+# Known displacements for these mapsequence layers when the layer index is set to 0
```
Comment (nabobalis) [inline sunkit_image/tests/test_coalignment.py:415]:
```suggestion
```

File: sunkit_image/tests/test_coalignment.py [hunk 12]
```diff
@@ -387,3 +392,90 @@ def test_apply_shifts(aia171_test_map):
...
+@pytest.fixture
+def known_displacements_layer_index0():
+    return {"x": np.asarray([0.0, -9.827465, -19.676442]), "y": np.asarray([0.0, 0.251137, 0.490014])}
+
+
+# Known displacements for these mapsequence layers when the layer index is set to 1
```
Comment (nabobalis) [inline sunkit_image/tests/test_coalignment.py:421]:
```suggestion
```

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
I will merge when the CI passes, with a new release as well?

**Pull request merged**